resolver/serviceResolver.go

```diff
@@ -269,6 +269,82 @@ func (resolver *serviceResolver) SignMessage(userIp string, request requests.Sig
 
 }
 
+// SetSecurityModeNoExpire gets the user's guardian, verifies the codes and then sets the SecurityMode
+func (resolver *serviceResolver) SetSecurityModeNoExpire(userIp string, request requests.SetSecurityModeNoExpireMessage) (*requests.OTPCodeVerifyData, error) {
+	userAddress, err := sdkData.NewAddressFromBech32String(request.UserAddr)
```

Comment:
a big part of these 2 functions is very similar, it can be extracted into a common function

Reply:
updated with a common function